bgpd/bgp_zebra.c

```diff
@@ -1425,9 +1427,14 @@ void bgp_zebra_announce(struct bgp_node *rn, struct prefix *p,
 			    && !CHECK_FLAG(api.flags, ZEBRA_FLAG_EVPN_ROUTE))
 				sprintf(label_buf, "label %u",
 					api_nh->labels[0]);
-			zlog_debug("  nhop [%d]: %s if %u VRF %u %s",
-				   i + 1, nh_buf, api_nh->ifindex,
-				   api_nh->vrf_id, label_buf);
+			if (CHECK_FLAG(api.flags, ZEBRA_FLAG_EVPN_ROUTE)
```

Comment:
the logic that uses `label_buf` only runs for _non_ evpn routes - it looks as if `label_buf` may be printed but not filled in?

Reply:
Indeed. Also, `snprintf()` should be preferred over `sprintf()` as a security best practice.

Reply:
@mjstapp `label_bug` is initialized with null terminator. In absence of label for a route it will display as space. 
`label_buf[0] = '\0';`

Reply:
@rwestphal  I have converted to snprintf for RMAC field.

Reply:
I see where label_buf is re-init'd for each nexthop - but it looks as if eth_buf is not being reset/cleared?

Reply:
Ack. I made the change to reset eth_buf within nexthop iteration.